sdk/operationalinsights/loganalytics/README.md

```diff
@@ -48,6 +49,37 @@ msRestNodeAuth.interactiveLogin().then((creds) => {
 });
 ```
 
+###### Client Credential Login
+```ts
+import * as msRest from "@azure/ms-rest-js";
+import * as msRestNodeAuth from "@azure/ms-rest-nodeauth";
+import { LogAnalyticsClient, LogAnalyticsModels } from "@azure/loganalytics";
+
+const tenantId = process.env["AZURE_TENANT_ID"];
+const clientId = process.env["AZURE_CLIENT_ID"];
+const clientSecret = process.env["AZURE_CLIENT_SECRET"];
+
+const credentials:msRest.ServiceClientCredentials = await msRestNodeAuth.loginWithServicePrincipalSecret(
+  clientId,
+  clientSecret,
+  tenantId,
+  {
+    tokenAudience: "https://api.loganalytics.io"
```

Comment:
Is this because without the tokenAudience set to this value, authentication does not work at all?

Reply:
Thanks @ramya-rao-a ,
As long as I investigated, when Client Credential is used for authentication without `tokenAudience`, 403 response and `InvalidTokenError` occur as the following.

This is mitigated by adding `tokenAudience: "https://api.loganalytics.io"`.

```
RestError: The provided authentication is not valid for this resource
    at new RestError (PATH\TO\PROJECT\node_modules\@azure\ms-rest-js\dist\msRest.node.js:2480:28)
    at PATH\TO\PROJECT\node_modules\@azure\ms-rest-js\dist\msRest.node.js:3687:37
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  code: 'InvalidTokenError',
  statusCode: 403,
  request: WebResource {
    streamResponseBody: false,
    url: 'https://api.loganalytics.io/v1/workspaces/WORKSPACE_ID/query',
    method: 'POST',
    headers: HttpHeaders { _headersMap: [Object] },
    body: '{"query":"Usage"}',
    query: undefined,
    formData: undefined,
    withCredentials: false,
    abortSignal: undefined,
    timeout: 0,
    onUploadProgress: undefined,
    onDownloadProgress: undefined,
    proxySettings: undefined,
    keepAlive: undefined,
    agentSettings: undefined,
    redirectLimit: undefined,
    operationSpec: {
      httpMethod: 'POST',
      path: 'v1/workspaces/{workspaceId}/query',
      urlParameters: [Array],
      requestBody: [Object],
      responses: [Object],
      serializer: [Serializer]
    }
  },
  response: {
    body: '{"error":{"message":"The provided authentication is not valid for this resource","code":"InvalidTokenError","correlationId":"c3a95fc8-dbcc-490c-95e5-d5fef83296b3","innererror":{"code":"InvalidTokenError","message":"Could not validate the request. Challenge failed: SignatureVerificationFailed"}}}',
    headers: HttpHeaders { _headersMap: [Object] },
    status: 403
  },
  body: {
    error: {
      code: 'InvalidTokenError',
      message: 'The provided authentication is not valid for this resource',
      innererror: [Object],
      correlationId: 'c3a95fc8-dbcc-490c-95e5-d5fef83296b3'
    }
  }
}
```

Reply:
As @ramya-rao-a wrote, I confirmed thatn `msRestNodeAuth.interactiveLogin()` in Interactive Login sample also needs `tokenAudience` as the following code.
```ts
msRestNodeAuth.interactiveLogin({tokenAudience: "https://api.loganalytics.io"}).then((creds) => {
  const client = new LogAnalyticsClient(creds);
```

Reply:
Thanks for the confirmation @horihiro 
In that case, I would recommend that we update the existing code snippet to take the tokenAudience rather than add a new code snippet

Reply:
@ramya-rao-a I changed sample code in this PR.